File: app/src/main/java/han/ayeon/searchimgandvideo/util/Converter.java
package han.ayeon.searchimgandvideo.util;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class Converter {

    private static SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.SSS", Locale.KOREA);

    public static Date StringToDate(String dateTime) {

        if(dateTime!=null) {
            String time = dateTime.replace("T", " ");
            Date date = null;
            try {
                date = formatter.parse(time);
            } catch (ParseException e) {
                e.printStackTrace();
            }
            return date;
        } else {
            return null;
        }


    }
}

File: app/src/main/java/han/ayeon/searchimgandvideo/util/Converter.kt
package han.ayeon.searchimgandvideo.util

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by HanAYeon on 2019-07-18.
 */

private val formatter = SimpleDateFormat("yyyy-MM-dd hh:mm:ss.SSS", Locale.KOREA)

fun String.toDate() : Date? {
    val time = this.replace("T", " ")
    try {
        return formatter.parse(time)
    } catch (e: ParseException) {
        e.printStackTrace()
    }

    return null
}